draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,52 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New events SHOULD be defined using an extension schema, using the annotations
+and concepts presented in this section.
+
+Each extension schema is named by a URI. That URI MUST be absolute; it precisely
+identifies the format and meaning of the extension. URIs that contain a domain
+name SHOULD also contain a month-date in the form mmyyyy. The definition of the
+element and assignment of the URI MUST have been authorized by the owner of the
+domain name on or very close to that date. (This avoids problems when domain
+names change ownership.) If the resource or document defines several categories
+or groups of events, then the URI MUST identify the actual extension in use,
+e.g., using a fragment or query identifier (characters after a "#" or "?" in the
+URI).
 
-This document defines the main schema for the qlog format together with some
-common events, which on their own do not provide much logging utility. It is
-expected that logging is extended with specific, per-protocol event definitions
-that specify the name (category + type) and data needed for each individual
-event. Examples include the QUIC event definitions {{QLOG-QUIC}} and HTTP/3
-event definitions {{QLOG-H3}}.
+For extensions defined in RFCs, the URI used SHOULD be a URN starting with
+`urn:ietf:params:qlog` followed by a registered, descriptive name.
 
-This section defines some basic annotations and concepts that SHOULD be used by
-event definition documents. Doing so ensures a measure of consistency that makes
-it easier for qlog implementers to support a wide variety of protocols.
+A log file that that uses events from extension schema SHOULD list all schema
+identifiers in the `additional_event_schema` field.
+
+For example, a qlog file that uses two hypothetical qlog extension schema named
+"rick" and "morty" would indicate this as:
+
+~~~
+additional_event_schema = [
+                            urn:ietf:params:qlog:rick,
+                            urn:ietf:params:qlog:morty
+                          ]
```

Comment:
This could do with an extension of the example to also show at least one domainname-based extension (i.e., things that are not defined in RFCs), with the mmyyyy date in the URI